Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -490,6 +616,26 @@ namespace mu2e {
     return retval;
   }
 
+  bool LoopHelixFit::goodHelix(HelixSeed const& hseed) const {
+    // check helicity.  The test on the charge and helicity
+    if(!hseed.status().hasAllProperties(goodseed_)) return false;
+    // test helix
+    auto const& helix = hseed.helix();
+    if(helix.radius() == 0.0 || helix.lambda() == 0.0)
+      throw cet::exception("RECO")<<"mu2e::HelixFit: degenerate seed parameters" << endl;
+    // test the helix slope significance if requested
+    if(useSlopeSigCut_) {
+      const float slope    = hseed.recoDir().slope();
+      const float slopeErr = std::fabs(hseed.recoDir().slopeErr());
+      const float slopeSig = (slopeErr > 0.f) ? slope/slopeErr : 0.f;
+      if(slopeSig*fdir_.dzdt() < slopeSigCut_) { //slope is below the given threshold relative to the given direction
+        if(print_ > 0) printf("[LoopHelixFit::%s] Skipping helix seed with slope significance %.1f\n", __func__, slopeSig);
+        return false;
```

Comment:
I am confused by the logic: if the slope isn't significant, don't we still want to fit the track? Especially if the direction is hard-coded.

Reply:
This was a request from Giani, that for the trigger we may want to skip fitting helix seeds that are very likely upstream tracks when only looking for downstream tracks. In the Offline processing, I imagine there's no harm in fitting these downstream as well since we're forcing a downstream fit for all tracks.

Reply:
I think the intent is:
- if the slope is significant, only fit that one (significant) direction
- if the slope isn't significant fit both directions
Maybe I've misunderstood the logic, but it seems the fit is skipped entirely if the slope isn't significant.

Reply:
This would only be used if using a forced fit direction, and throws an error if used in the helix slope-based fit direction mode. The intent with this is in the Online processing, only fit with the forced downstream hypothesis and skip events that have a significant upstream slope to save processing time with minimal efficiency loss for downstream tracks. This update is essentially separate from the rest of the updates as it's only relevant to the forced fit direction processing

Reply:
I'm starting to think this test belongs outside loopHelixFit, either in the helix fitter itself, or (after merging) in a dedicated module that would only be run in the trigger reco sequence.

Reply:
That's a good point. Perhaps this should be a new producer module that reads in a helix seed collection and produces a filtered helix collection? This way you can tailor the collection for the LoopHelixFit input, but not lose these helices if another path may want to use them in a separate LoopHelixFit module instance (e.g. an upstream trigger that only looks at significantly upstream-oriented helices using the same initial helix collection).

Reply:
I moved this filtering to the HelixFilter module, which seemed a more appropriate place for this type of selection.